python/ray/_common/pydantic_compat.py

```diff
@@ -106,3 +106,28 @@ def register_pydantic_serializers(serialization_context):
         },
         custom_deserializer=lambda kwargs: ModelField(**kwargs),
     )
+
+
+def model_dump_json(model: BaseModel) -> dict:
+    if IS_PYDANTIC_2:
+        return model.model_dump_json()
+    else:
+        return model.json()
+
+
+def from_json(json_str: str) -> BaseModel:
```

Comment:
```suggestion
def from_json(json_str: str) -> Any:
```

Reply:
can we make this model_validate_json(json_str) instead? and skip needing to do obj_to_model(from_json(str))?